### DISCUSSION TITLE
babel-plugin-optimize-clsx equivalent for swc

### DISCUSSION THREAD

**Discussion by kripod** [vercel/next.js#30365] [Ideas]
### Describe the feature you'd like to request

Utility-based CSS libraries like [Tailwind](https://tailwindcss.com/) rely heavily upon class composition. There are libraries out there which provide a convenient interface for composing class names conditionally, e.g.:

- [classnames](https://github.com/JedWatson/classnames)
- [clsx](https://github.com/lukeed/clsx)
- [classcat](https://github.com/jorgebucaran/classcat)

[A Babel plugin](https://github.com/merceyz/babel-plugin-optimize-clsx) was born to replace class concatenation method calls with inlined code.

### Describe the solution you'd like

As mentioned by the [Next.js 12 announcement post](https://nextjs.org/blog/next-12#faster-builds-and-fast-refresh-with-rust-compiler), custom swc transforms could be introduced.

### Describe alternatives you've considered

- Keeping the Babel config as-is, compromising on compilation performance
- Removing the Babel config in its entirely, compromising on end-user performance (bundle size & code execution speed)

**Comment by timneutkens** [comment]
Please post this here so that it's all tracked in the same thread: https://github.com/vercel/next.js/discussions/30174
